service-registry/src/main/java/org/apache/servicecomb/serviceregistry/client/http/ServiceRegistryClientImpl.java

```diff
@@ -142,6 +143,23 @@ private void retry(RequestContext requestContext, Handler<RestResponse> response
               return;
             }
 
+            // no need to generate warn log when schema or service not exist
+            if (HttpStatusClass.CLIENT_ERROR.equals(HttpStatusClass.valueOf(response.statusCode()))) {
+              try {
+                @SuppressWarnings("unchecked")
+                String errorCode = JsonUtils.readValue(bodyBuffer.getBytes(), Map.class).get("errorCode").toString();
+                // 400012: ErrServiceNotExists
+                // 400016: ErrSchemaNotExists
+                if (errorCode.equals("400012") || errorCode.equals("400016")) {
+                  countDownLatch.countDown();
+                  return;
+                }
+              } catch (IOException | NullPointerException ignored) {
+                // Ignore IOException when read value from bodyBuffergit
+                // Ignore NullPointerException when bodyBuffer not contains errorCode
```

Comment:
Does this mean these two exception will never happen? I don't think it's not a good idea to ingore them when the exception will throw.

Reply:
IOException occures when JsonUtil read value failed and NullPointerException occurs when bodyBuffer not contains errorCode. I will log some warn info to deal with them.

Reply:
Already modified as suggested